### ISSUE TITLE
DKImagePickerController problems when compiling

### ISSUE THREAD

**Issue by wobfan** [miguelpruivo/flutter_file_picker#966]
**Describe the bug**

A clear and concise description of what the bug is. If the issue happens to be on Android, please make sure that it also happens with a different device/simulator and/or version.
`flutter build ipa` and `flutter build ios` fail with compilation errors.

Before that, I got problems with the `DKImagePickerController-DKImagePickerController` and `DKPhotoGallery-DKPhotoGallery` pods. I need to pick a `Team` under `Signing & Capabilities` everytime before I build or run. 

**Platform**

- [ ] Android
- [x] iOS
- [ ] Web
- [ ] Desktop

**Platform OS version**
macOS 12.1

**Details to reproduce the issue**
I cannot remember doing anything else than adding the `file_picker` plugin to Flutter. With that, the above mentioned problems came. I needed to set a Team in Xcode for both plugins, and then I got this error.
It's weird, because it seems to be a more deep error in a library ([DKImagePickerController](https://github.com/zhangao0086/DKImagePickerController)) that `file_picker` uses.

**Error Log**
```
               SwiftCodeGeneration normal arm64 (in target 'DKImagePickerController' from project 'Pods')
                   cd /Users/wobbe/workspace/app/FlutterFrontend/ios/Pods
                   /Applications/Xcode-beta.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swift-frontend -frontend -c -primary-file /Users/wobbe/Library/Developer/Xcode/DerivedData/Runner-evqxwyagqpvqfwcvtzybvxrfqkvk/Build/Intermediates.noindex/ArchiveIntermediates/Runner/IntermediateBuildFilesPath/Pods.build/Release-iphoneos/DKImagePickerController.build/Objects-normal/arm64/DKPopoverViewController.bc -embed-bitcode -target arm64-apple-ios9.0 -Xllvm -aarch64-use-tbi -O -disable-llvm-optzns -module-name DKImagePickerController -o /Users/wobbe/Library/Developer/Xcode/DerivedData/Runner-evqxwyagqpvqfwcvtzybvxrfqkvk/Build/Intermediates.noindex/ArchiveIntermediates/Runner/IntermediateBuildFilesPath/Pods.build/Release-iphoneos/DKImagePickerController.build/Objects-normal/arm64/DKPopoverViewController.o
               LLVM ERROR: out of memory
               Allocation failed
               Please submit a bug report (https://swift.org/contributing/#reporting-bugs) and include the project and the crash backtrace.
               Stack dump:
               0.	Program arguments: /Applications/Xcode-beta.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swift-frontend -frontend -c -primary-file /Users/wobbe/Library/Developer/Xcode/DerivedData/Runner-evqxwyagqpvqfwcvtzybvxrfqkvk/Build/Intermediates.noindex/ArchiveIntermediates/Runner/IntermediateBuildFilesPath/Pods.build/Release-iphoneos/DKImagePickerController.build/Objects-normal/arm64/DKPopoverViewController.bc -embed-bitcode -target arm64-apple-ios9.0 -Xllvm -aarch64-use-tbi -O -disable-llvm-optzns -module-name DKImagePickerController -o /Users/wobbe/Library/Developer/Xcode/DerivedData/Runner-evqxwyagqpvqfwcvtzybvxrfqkvk/Build/Intermediates.noindex/ArchiveIntermediates/Runner/IntermediateBuildFilesPath/Pods.build/Release-iphoneos/DKImagePickerController.build/Objects-normal/arm64/DKPopoverViewController.o
               1.	Apple Swift version 5.6 (swiftlang-5.6.0.322.3 clang-1316.0.18.9)
               2.	Compiling with the current language version
               3.	Running pass 'Function Pass Manager' on module '/Users/wobbe/Library/Developer/Xcode/DerivedData/Runner-evqxwyagqpvqfwcvtzybvxrfqkvk/Build/Intermediates.noindex/ArchiveIntermediates/Runner/IntermediateBuildFilesPath/Pods.build/Release-iphoneos/DKImagePickerController.build/Objects-normal/arm64/DKPopoverViewController.bc'.
               4.	Running pass 'ObjC ARC contraction' on function '@UI_USER_INTERFACE_IDIOM'
               Stack dump without symbol names (ensure you have llvm-symbolizer in your PATH or set the environment var `LLVM_SYMBOLIZER_PATH` to point to it):
               0  swift-frontend           0x0000000112b97577 llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) + 39
               1  swift-frontend           0x0000000112b965c8 llvm::sys::RunSignalHandlers() + 248
               2  swift-frontend           0x0000000112b97bd0 SignalHandler(int) + 288
               3  libsystem_platform.dylib 0x00007ff80ccd4e2d _sigtramp + 29
               4  libsystem_malloc.dylib   0x00007ff80caf4ad6 _malloc_zone_malloc + 125
               5  libsystem_c.dylib        0x00007ff80cc0bd10 abort + 123
               6  swift-frontend           0x0000000112ae475a llvm::report_bad_alloc_error(char const*, bool) + 106
               7  swift-frontend           0x0000000112ae4772 out_of_memory_new_handler() + 18
               8  libc++abi.dylib          0x00007ff80cc7d9bb operator new(unsigned long) + 43
               9  swift-frontend           0x00000001128a59bd llvm::Function::BuildLazyArguments() const + 77
               10 swift-frontend           0x000000011068f487 llvm::objcarc::BundledRetainClaimRVs::insertRVCallWithColors(llvm::Instruction*, llvm::CallBase*, llvm::DenseMap<llvm::BasicBlock*, llvm::TinyPtrVector<llvm::BasicBlock*>, llvm::DenseMapInfo<llvm::BasicBlock*>, llvm::detail::DenseMapPair<llvm::BasicBlock*, llvm::TinyPtrVector<llvm::BasicBlock*> > > const&) + 151
               11 swift-frontend           0x00000001106a2588 (anonymous namespace)::ObjCARCContract::run(llvm::Function&, llvm::AAResults*, llvm::DominatorTree*) + 1384
               12 swift-frontend           0x00000001128dce20 llvm::FPPassManager::runOnFunction(llvm::Function&) + 1488
               13 swift-frontend           0x00000001128e3b13 llvm::FPPassManager::runOnModule(llvm::Module&) + 67
               14 swift-frontend           0x00000001128dd5d9 llvm::legacy::PassManagerImpl::run(llvm::Module&) + 1161
               15 swift-frontend           0x000000010ddff2ff swift::performLLVMOptimizations(swift::IRGenOptions const&, llvm::Module*, llvm::TargetMachine*) + 3791
               16 swift-frontend           0x000000010de004ec swift::performLLVM(swift::IRGenOptions const&, swift::DiagnosticEngine&, llvm::sys::SmartMutex<false>*, llvm::GlobalVariable*, llvm::Module*, llvm::TargetMachine*, llvm::StringRef, swift::UnifiedStatsReporter*) + 2812
               17 swift-frontend           0x000000010de096c5 swift::performLLVM(swift::IRGenOptions const&, swift::ASTContext&, llvm::Module*, llvm::StringRef) + 213
               18 swift-frontend           0x000000010d8c4f95 swift::performFrontend(llvm::ArrayRef<char const*>, char const*, void*, swift::FrontendObserver*) + 16565
               19 swift-frontend           0x000000010d883de8 swift::mainEntry(int, char const**) + 1064
               20 dyld                     0x00000001216b94fe start + 462
               error: Abort trap: 6 (in target 'DKImagePickerController' from project 'Pods')

```

**Screenshots and/or video**
If applicable, add screenshots or video to help explain your problem.

**Flutter Version details**
```
11:39:27 ~ > flutter doctor -v
[✓] Flutter (Channel stable, 2.10.1, on macOS 12.1 21C52 darwin-x64, locale
    en-DE)
    • Flutter version 2.10.1 at /usr/local/Caskroom/flutter/2.8.1/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision db747aa133 (7 days ago), 2022-02-09 13:57:35 -0600
    • Engine revision ab46186b24
    • Dart version 2.16.1
    • DevTools version 2.9.2

[!] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
    • Android SDK at /Users/wobbe/Library/Android/sdk
    • Platform android-32, build-tools 32.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)
    ! Some Android licenses not accepted.  To resolve this, run: flutter doctor --android-licenses

[✓] Xcode - develop for iOS and macOS (Xcode 13.3)
    • Xcode at /Applications/Xcode-beta.app/Contents/Developer
    • CocoaPods version 1.11.2

[✗] Chrome - develop for the web (Cannot find Chrome executable at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome)
    ! Cannot find Chrome. Try setting CHROME_EXECUTABLE to a Chrome executable.

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)

[!] Connected device
    ! No devices available

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 3 categories.
```

I imagine that this problem is not very related to `file_picker` but as it came up after I added this plugin I hope that someone has a clue where to look to fix the problem, or maybe also other people experience the same.

**Comment by Erchil66** [comment]
@wobfan we did you already solve this? am also stuck inhere after upgrading in xcode 13.3 beta 2 then after that same as you.

do we have a solution for this @miguelpruivo

**Comment by wobfan** [comment]
Actually yes, @Erchil66.

In my case, i upgraded Xcode to 13.3 beta 2 in order to debug on my iPhone with iOS 15.4 beta 2.
After I installed Xcode beta and switched via `xcode-select -s /Applications/Xcode 13.3 beta 2` I was able to access the iPhone for debugging but the error occured. I didn't think of the beta being a problem then, but out of curiosity I switched back to the original Xcode 13.2 via `xcode-select ...`. I could then still access my beta-iOS iPhone for debugging (Idk why), and the error disappeared.

So, TLDR: switched back to non-beta Xcode.

**Comment by Erchil66** [comment]
@wobfan, Thanks.

i also go back to 13.2 then the appstore luckily also need update after that it's now compiling and running.

Yes, backing to non-beta xcode also works on me really thanks

**Comment by miguelpruivo** [comment]
Hi! I'm happy that you've managed to solve it, only had the opportunity to check this issue now.

I can't ensure Xcode beta support, specially when the plugin relies on other dependencies such as `DKImagePickerController` which might have some limitation with beta versions as well. If that happens, I suggest you to open an issue directly on [its repo](https://github.com/zhangao0086/DKImagePickerController).

Thank you!

**Comment by kazume** [comment]
Same fault on Xcode 13.3 (non-beta) now ... had to revert to 13.2.1 to be able to archive.
